src/NServiceBus.Core/Recoverability/Faults/MessageToBeRetried.cs

```diff
@@ -7,12 +7,13 @@ class MessageToBeRetried : MessageProcessingFailed
     {
         public int Attempt { get; }
         public TimeSpan Delay { get; }
-        public bool IsImmediateRetry => Delay == TimeSpan.Zero;
+        public bool IsImmediateRetry { get; }
 
-        public MessageToBeRetried(int attempt, TimeSpan delay, IncomingMessage message, Exception exception) : base(message, exception)
+        public MessageToBeRetried(int attempt, TimeSpan delay, bool immediateRetry, ErrorContext errorContext) : base(errorContext)
         {
             Attempt = attempt;
             Delay = delay;
+            IsImmediateRetry = immediateRetry;
```

Comment:
I added a specific bool flag to distinguish between delayed and immediate retries as this caused wrong behavior when sending ready messages in case you configure a 0 delay for delayed retries. Although that config is suspicious too, it would not send a ready message although the message will be sent to the timeout manager, causing again a ready message loss.

I think we should probably disallow TimeSpan.Zero configs for delayed retries. But at least we need to be able to distinguish between immediate and delayed retries regardless. Surely there are better approaches. Feel free to refactor this, as I just wanted to fix the behavior